lib/pbench/server/api/resources/audit_query.py

```diff
@@ -0,0 +1,91 @@
+from http import HTTPStatus
+from logging import Logger
+
+from flask.json import jsonify
+from flask.wrappers import Request, Response
+
+from pbench.server import OperationCode, PbenchServerConfig
+from pbench.server.api.resources import (
+    APIAbort,
+    ApiAuthorizationType,
+    ApiBase,
+    ApiContext,
+    ApiMethod,
+    ApiParams,
+    ApiSchema,
+    Parameter,
+    ParamType,
+    Schema,
+)
+from pbench.server.database.models.audit import (
+    Audit,
+    AuditReason,
+    AuditSqlError,
+    AuditStatus,
+    AuditType,
+)
+
+
+class AuditQuery(ApiBase):
+    """
+    API class to retrieve and mutate server configuration settings.
+    """
+
+    def __init__(self, config: PbenchServerConfig, logger: Logger):
+        super().__init__(
+            config,
+            logger,
+            ApiSchema(
+                ApiMethod.GET,
+                OperationCode.READ,
+                query_schema=Schema(
+                    Parameter("dataset", ParamType.DATASET),
+                    Parameter("end", ParamType.DATE),
+                    Parameter("name", ParamType.STRING),
+                    Parameter("object_id", ParamType.STRING),
+                    Parameter("operation", ParamType.KEYWORD, enum=OperationCode),
+                    Parameter("reason", ParamType.KEYWORD, enum=AuditReason),
+                    Parameter("start", ParamType.DATE),
+                    Parameter("status", ParamType.KEYWORD, enum=AuditStatus),
+                    Parameter("type", ParamType.KEYWORD, enum=AuditType),
+                    Parameter("user", ParamType.USER),
+                    Parameter("user_id", ParamType.STRING),
+                    # FIXME: Implement pagination
+                    Parameter("offset", ParamType.INT),
+                    Parameter("limit", ParamType.INT),
+                ),
+                # FIXME: We should restrict non-ADMIN queries to audit records
```

Comment:
Good point, great item for a follow-up tracking issue.

Reply:
> Another option would be to simply require ADMIN role.

That's certainly where I would go, at least initially.